### PR TITLE
fix(driver/bpf): close maps on cleanup

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

/area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

Some internal maps of the bpf engine were not closed on cleanup. This fixes it.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libscap/engine/bpf/scap_bpf.c [hunk 1]
```diff
@@ -1384,6 +1384,11 @@ int32_t scap_bpf_close(struct scap_engine_handle engine)
 		handle->program_fd = -1;
 	}
 
+	for(int i = 0; i < BPF_MAPS_MAX; i++)
+	{
+		close(handle->m_bpf_map_fds[i]);
```
Comment (FedeDP) [inline userspace/libscap/engine/bpf/scap_bpf.c:1389]: Great catch! Can we check that each `fds > 0` too?
Also, i'd reset each entry to -1 after!

Comment (therealbobo) [reply userspace/libscap/engine/bpf/scap_bpf.c:1389]: Not all maps are populated, so I expect some of the results to be `-1`. The reset seems good to me.

File: userspace/libscap/engine/bpf/scap_bpf.c [hunk 2]
```diff
@@ -1384,6 +1384,12 @@ int32_t scap_bpf_close(struct scap_engine_handle engine)
...
```
Comment (FedeDP) [inline userspace/libscap/engine/bpf/scap_bpf.c:1389]:
```suggestion
                if (handle->m_bpf_map_fds[i] > 0) 
                {
		    close(handle->m_bpf_map_fds[i]);
		}
```

Comment (therealbobo) [reply userspace/libscap/engine/bpf/scap_bpf.c:1389]: Oh, I see now.

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**